### PR TITLE
Arnej/storage cluster context

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@baldersheim
@geirst
@vekterli 

please review

### CODE REVIEW

File: storage/src/tests/distributor/dummy_cluster_context.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#pragma once
```
Comment (geirst) [inline storage/src/tests/distributor/dummy_cluster_context.h:1]: Please add copyright

File: storage/src/vespa/storage/common/cluster_context.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/vespalib/stllike/string.h>
+
+namespace storage {
+
+struct ClusterContext {
```
Comment (geirst) [inline storage/src/vespa/storage/common/cluster_context.h:9]: Please add class comment.

File: storage/src/vespa/storage/persistence/bucketownershipnotifier.cpp [hunk 3]
```diff
@@ -59,7 +59,8 @@ BucketOwnershipNotifier::sendNotifyBucketToDistributor(
     }
     auto notifyCmd = std::make_shared<api::NotifyBucketChangeCommand>(bucket, infoToSend);
 
-    notifyCmd->setAddress(api::StorageMessageAddress::create(&_component.getClusterName(), lib::NodeType::DISTRIBUTOR, distributorIndex));
+    auto cluster_np = _component.cluster_context().cluster_name_ptr();
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/bucketownershipnotifier.cpp:62]: Consider `const auto*`

File: storage/src/vespa/storage/storageserver/mergethrottler.cpp [hunk 4]
```diff
@@ -373,7 +373,8 @@ MergeThrottler::forwardCommandToNode(
                     mergeCmd.getMaxTimestamp(),
                     mergeCmd.getClusterStateVersion(),
                     newChain));
-    fwdMerge->setAddress(api::StorageMessageAddress::create(&_component.getClusterName(), lib::NodeType::STORAGE, nodeIndex));
+    auto cluster_np = _component.cluster_context().cluster_name_ptr();
```
Comment (vekterli) [inline storage/src/vespa/storage/storageserver/mergethrottler.cpp:376]: Consider `const auto*`

File: storage/src/vespa/storage/common/cluster_context.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+protected:
+    virtual ~ClusterContext() = default;
+public:
+    // returns a pointer to the cluster name
+    // must be a valid pointer to a constant string for the
+    // lifetime of all the components that may ask for it
+    virtual const vespalib::string * cluster_name_ptr() const = 0;
```
Comment (vekterli) [inline storage/src/vespa/storage/common/cluster_context.h:16]: Consider adding a `noexcept` specifier to this

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**